src/main/java/uk/gov/companieshouse/overseasentitiesapi/controller/CostsController.java

```diff
@@ -47,12 +51,21 @@ public ResponseEntity<List<Cost>> getCosts(
         ApiLogger.infoContext(requestId, "Calling CostsService to retrieve costs", logMap);
 
         try {
-            var cost = costsService.getCosts(requestId, overseasEntityId);
+            Cost cost;
+            if (isRoeUpdateEnabled) {
+                cost = costsService.getCosts(requestId, overseasEntityId);
+            } else {
+                cost = costsService.getCostsForRegistration();
+            }
 
             return ResponseEntity.ok(Collections.singletonList(cost));
         } catch (SubmissionNotFoundException e) {
             ApiLogger.errorContext(requestId, "Error determining submission costs", e, logMap);
             return new ResponseEntity<>(HttpStatus.INTERNAL_SERVER_ERROR);
         }
     }
+
```

Comment:
We don't need this 'set' method, you can just make use of ReflectionTestUtils in the unit-test to set the value to what you want. E.g.

        ReflectionTestUtils.setField(costController, "isRoeUpdateEnabled", value);

See the `OverseasEntitiesControllerTest` for a working example.

Reply:
Happy to change that but not too fond of using reflection for this but as it's already used.